src/main/groovy/pl/allegro/tech/build/axion/release/infrastructure/git/GitRepository.groovy

```diff
@@ -168,6 +169,50 @@ class GitRepository implements ScmRepository {
                 repository.branch.current.name
         )
     }
+		
+		boolean isHeadCommit(String commitId) {
+			if(commitId?.trim()) {
+				ObjectId headId = repository.repository.jgit.repository.resolve(Constants.HEAD)
+				return Objects.equals(ObjectId.fromString(commitId), headId)
+			}
+			return false;
+		}
+		
+		LinkedHashMap<String, List<String>> allTaggedCommits(Pattern pattern, String maybeSinceCommit, boolean inclusive) {
```

Comment:
how about returning `List<TagsOnCommit>` ? `TagsOnCommit` is precisely what you are trying to achieve via `String -> List<String>` mapping; it also has `isHead` information, so you don't have to expose & call `isHead` method from the resolver